src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,242 @@
 import com.segment.analytics.integrations.Integration;
 import com.segment.analytics.integrations.Logger;
-import com.segment.analytics.integrations.ScreenPayload;
 import com.segment.analytics.integrations.TrackPayload;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Date;
+
 import static com.segment.analytics.internal.Utils.hasPermission;
 import static com.segment.analytics.internal.Utils.isNullOrEmpty;
-import static com.segment.analytics.Analytics.LogLevel.VERBOSE;
+import static com.segment.analytics.internal.Utils.toISO8601Date;
 
 /**
- * Google Analytics for Firebase is a free app measurement solution that provides insight on
- * app usage and user engagement.
+ * Google Analytics for Firebase is a free app measurement solution that provides insight on app
+ * usage and user engagement.
  *
  * @see <a href="https://firebase.google.com/docs/analytics/">Google Analytics for Firebase</a>
  * @see <a href="#">Google Analytics for Firebase Integration</a>
  * @see <a href="#">Google Analytics for Firebase Android SDK</a>
  */
 public class FirebaseIntegration extends Integration<FirebaseAnalytics> {
 
-  public static FirebaseAnalytics mFirebaseAnalytics;
-
-  public static final Factory FACTORY = new Factory() {
-    @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-      Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
-      if (!hasPermission(analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
-        logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
-        return null;
-      }
-      if (!hasPermission(analytics.getApplication(), Manifest.permission.INTERNET)) {
-        logger.debug("INTERNET is required for Firebase Analytics.");
-        return null;
-      }
+  public static final Factory FACTORY =
+      new Factory() {
+        @Override
+        public Integration<?> create(ValueMap settings, Analytics analytics) {
+          Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
+          if (!hasPermission(
+              analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
+            logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
+            return null;
+          }
+          if (!hasPermission(analytics.getApplication(), Manifest.permission.WAKE_LOCK)) {
+            logger.debug("WAKE_LOCK is required for Firebase Analytics.");
+            return null;
+          }
+
+          Context context = analytics.getApplication();
+
+          return new FirebaseIntegration(context, logger);
+        }
+
+        @Override
+        public String key() {
+          return FIREBASE_ANALYTICS_KEY;
+        }
+      };
+
+  private static final String FIREBASE_ANALYTICS_KEY = "Firebase";
+  final Logger logger;
+  final FirebaseAnalytics firebaseAnalytics;
+  private static final Map<String, String> eventMapper = createEventMap();
```

Comment:
Constants should be constant cased https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names.

Reply:
i.e. this variable should be `EVENT_MAPPER`.